crypto/pkcs7/pk7_lib.c

```diff
@@ -28,6 +28,11 @@ long PKCS7_ctrl(PKCS7 *p7, int cmd, long larg, char *parg)
     /* NOTE(emilia): does not support detached digested data. */
     case PKCS7_OP_SET_DETACHED_SIGNATURE:
         if (nid == NID_pkcs7_signed) {
+            if (p7->d.sign == NULL) {
+                ERR_raise(ERR_LIB_PKCS7, PKCS7_R_NO_CONTENT);
```

Comment:
I'm not sure this should raise an error here - there is no error raised if p7->d.sign == NULL in the PKCS7_OP_GET_DETACHED_SIGNATURE case - just a ret = 1 return.

The return processing here is a little odd too - in that in all other cases the control returns 0 on error and 1 on success but in this one it returns whatever the user passed in - and that seems strange indeed.

With no official docs for this API, whether or not this makes sense is unclear

Reply:
It is actually debatable what return value should be returned on PKCS7_OP_GET_DETACHED_SIGNATURE with p7->d.sign == NULL - as if there is no content there is no signature (detached or not).

IMO raising the error on SET makes sense.